03_Modules/Transactions/src/module/CostNotifier.ts

```diff
@@ -2,12 +2,9 @@ import { ITransactionEventRepository } from '@citrineos/data';
 import { AbstractModule, CallAction } from '@citrineos/base';
 import { ILogObj, Logger } from 'tslog';
 import { CostCalculator } from './CostCalculator';
+import { Scheduler } from './Scheduler';
 
-export class CostNotifier {
-  private _registry: Map<string, NodeJS.Timeout> = new Map();
-
-  private readonly _logger: Logger<ILogObj>;
-
+export class CostNotifier extends Scheduler {
```

Comment:
Questioning if we really need to extend from Scheduler.

Reply:
With the current set up, if you try to unit test CostNotifier, won't you inadvertently be testing `Scheduler` due to the inheritance?

Reply:
I think my preference would be that 1) there is no inheritance, and 2) `Scheduler` and `CostNotifier` each have there own unit tests.

My argument against inheritance is that we will be inadvertently testing inherited methods when we set up unit tests for any class that extends `Scheduler`. So if something in `Scheduler` breaks, then it will break unit tests for all classes that inherit from it.

Reply:
@langiole why no inheritance? Seems like a nice generic base functionality to have, and then extend with explicit implementations. Would you rather have a Scheduler class instantiated at start up and then passed down to the CostNotifier?

Reply:
That is what I was thinking. Like Damian said, composition over inheritance.

My reason against inheritance is that you will inadvertently be testing the `Scheduler` when you write unit test for any classes that inherit it (e.g. `CostNotifier`). In other words, you cannot unit test `CostNotifier` without inadvertently testing `Scheduler`.

That being said, I'm fine with keeping it since it makes sense conceptually.